util/template/expression_template.go

```diff
@@ -31,10 +33,13 @@ func expressionReplace(w io.Writer, expression string, env map[string]interface{
 	if _, ok := env["retries"]; !ok && hasRetries(unmarshalledExpression) && allowUnresolved {
 		// this is to make sure expressions like `sprig.int(retries)` don't get resolved to 0 when `retries` don't exist in the env
 		// See https://github.com/argoproj/argo-workflows/issues/5388
+		log.WithError(err).Debug("unresolved is allowed ")
 		return w.Write([]byte(fmt.Sprintf("{{%s%s}}", kindExpression, expression)))
 	}
 	result, err := expr.Eval(unmarshalledExpression, env)
-	if (err != nil || result == nil) && allowUnresolved { //  <nil> result is also un-resolved, and any error can be unresolved
+	if (err != nil || result == nil) && allowUnresolved {
+		//  <nil> result is also un-resolved, and any error can be unresolved
+		log.WithError(err).Debug("unresolved is allowed ")
```

Comment:
how do we know if we hit this branch, not line 26? Mayb add more message?

Reply:
When I look at the logs, I should be able to know which line of code the logging came from. If each message was different, that would work.